src/react/components/Icon.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2021 Ispirata Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import React from 'react';
+import { OverlayTrigger, Tooltip } from 'react-bootstrap';
+
+const iconToClassName = {
+  add: 'fas fa-plus',
+  arrowLeft: 'fas fa-chevron-left',
+  blocks: 'fas fa-shapes',
+  copyPaste: 'fas fa-paste',
+  delete: 'fas fa-times',
+  devices: 'fas fa-cube',
+  documentation: 'fa fa-book',
+  edit: 'fas fa-pencil-alt',
+  erase: 'fas fa-eraser',
+  filter: 'fas fa-filter',
+  flows: 'fas fa-wind',
+  groups: 'fas fa-object-group',
+  home: 'fas fa-home',
+  interfaces: 'fas fa-stream',
+  logout: 'fas fa-sign-out-alt',
+  pipelines: 'fas fa-code-branch',
+  settings: 'fas fa-cog',
+  statusOK: 'fas fa-check-circle color-green',
+  statusConnected: 'fas fa-circle color-green',
+  statusDisconnected: 'fas fa-circle color-red',
+  statusKO: 'fas fa-times-circle color-red',
+  statusNeverConnected: 'fas fa-circle color-grey',
+  triggers: 'fas fa-bolt',
+};
+
+type Icon = keyof typeof iconToClassName;
+
+interface Props {
+  className?: string;
+  icon: Icon;
+  onClick?: (event: React.MouseEvent<HTMLElement, MouseEvent>) => void;
+  style?: React.CSSProperties;
+  tooltip?: string;
+  tooltipPlacement?: React.ComponentProps<typeof OverlayTrigger>['placement'];
+}
+
+export default ({
+  className,
+  icon,
+  onClick,
+  style = {},
+  tooltip,
+  tooltipPlacement,
+}: Props): React.ReactElement => {
+  let iconClassName = iconToClassName[icon];
+  let iconStyle = {};
+
+  if (onClick) {
+    iconStyle = { ...iconStyle, cursor: 'pointer' };
+    if (icon === 'delete') {
```

Comment:
you could check the "onClick" property, instead of the actual property name to determine if it's a button or not

Reply:
The issue here is that some icons have an `onClick` prop but we're not displaying them as `.btn` buttons: `erase` and `edit` for instance.
As of now, the only exception is the `delete` icon that is always rendered as a `.btn .btn-danger` button (it always has an `onClick` prop).
So a slight improvement could be to modify the `delete` icon definition to always include `.btn .btn-danger`, but I am not convinced by that either.
Any clever ideas?

Reply:
If that's the case, i'd rather have a property that allows developers to decide how to render it.

```
type display = 'plain' | 'button' | 'link';
```
you can then check if the display is `button` and adjust the classes accordingly. By using clever defaults you should only modify the `delete` icon code

Reply:
I reworked this part, see if it resonates with you.
Now:
- there is a `as = 'default' | 'button'` prop to change the rendering behavior
- the list of dangerous icons is clearly stated and is styled automatically (with `.color-red` or `.btn-danger` as appropriate)